### PR TITLE
TS-4610: Fix HTTP and HTTP2 stats

### PR DESCRIPTION
**Pull request by shinrich**

### CODE REVIEW

**Comment by atsci** [comment]
FreeBSD build *successful*! See https://ci.trafficserver.apache.org/job/Github-FreeBSD/378/ for details.

**Comment by atsci** [comment]
Linux build *successful*! See https://ci.trafficserver.apache.org/job/Github-Linux/274/ for details.

File: proxy/http/HttpSM.cc [hunk 1]
```diff
@@ -200,7 +200,8 @@ HttpVCTable::cleanup_entry(HttpVCTableEntry *e)
     // Update stats
     switch (e->vc_type) {
     case HTTP_UA_VC:
-      //              HTTP_DECREMENT_DYN_STAT(http_current_client_transactions_stat);
+      // Moved to HttpSM::destroy
+      // HTTP_DECREMENT_DYN_STAT(http_current_client_transactions_stat);
```
Comment (zwoop) [inline proxy/http/HttpSM.cc]: I don't know if we need to keep these commented out, but maybe  just a comment saying "Metrics are updated in ::destroy".

**Comment by zwoop** [comment]
So, I tried this patch, and it does not fix the problem I'm running into, where proxy.process.http.current_active_client_connections is seemingly never decremented. I had a merge conflict, but I'm pretty sure I corrected it right.

**Comment by zwoop** [comment]
Some sort of odd race condition here maybe. If I run single requests using curl, I'm not seeing the problem. But if I blast it with "ab -c 10 -n 10000" the counter permanently bumps up by 10,000.

**Comment by shinrich** [comment]
Yes, that fix does not address active client connection. It wasn't immediately clear to me what that meant for HTTP2. So I fixed the others but left active client connections alone.

On Tue, Jul 05, 2016 at 8:05 PM, Leif Hedstrom < notifications@github.com [notifications@github.com] > wrote:
So, I tried this patch, and it does not fix the problem I'm running into, where proxy.process.http.current_active_client_connections is seemingly never decremented. I had a merge conflict, but I'm pretty sure I corrected it right.

**Comment by zwoop** [comment]
@shinrich @masaori335  Should we land this patch though? I'm landing my other fix for the active client conns.

**Comment by masaori335** [comment]
@zwoop Looks good. We should land this to add a option to get metrics per protocol.

**Comment by zwoop** [comment]
It does have merge conflicts, so maybe I should resolve that locally, and commit it with --author=susan ? @PSUdaemon  wdyt?

**Comment by zwoop** [comment]
I resolved the conflicts, I think: http://paste.fedoraproject.org/389174/80381301/

**Comment by PSUdaemon** [comment]
Yes, merge and mark what files you resolved conflicts in. Git normally puts this in the commented part of the commit message so just make sure you uncomment before commit.

**Comment by zwoop** [comment]
This might be above my Git skill level. I did try to pull the PR into my tree, but it's so far behind that it made some crazy changes / conflicts :-/. Manually applying the "diff" from this PR, and fix the conflicts was much more doable.

**Pull request closed without merging**